runtime/src/iree-amd-aie/driver/xrt/nop_executable_cache.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_AMD_AIE_DRIVER_XRT_NOP_EXECUTABLE_CACHE_H_
+#define IREE_AMD_AIE_DRIVER_XRT_NOP_EXECUTABLE_CACHE_H_
+
+#include "iree/base/api.h"
+#include "iree/hal/api.h"
+#include "xrt/xrt_device.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Creates a no-op executable cache that does not cache at all.
+// This is useful to isolate pipeline caching behavior and verify compilation
+// behavior.
+//
+// |out_executable_cache| must be released by the caller (see
+// iree_hal_executable_cache_release).
+iree_status_t iree_hal_xrt_nop_executable_cache_create(
+    xrt::device device, iree_allocator_t host_allocator,
```

Comment:
When we have `host_allocator`, the convention is to place it right before what it allocates, that is `out_executable_cache`.